src/Microsoft.ML.Data/Transforms/ColumnConcatenatingTransformer.cs

```diff
@@ -119,14 +120,16 @@ public Arguments(string name, params string[] source)
             public Column[] Columns;
         }
 
-        public sealed class TaggedArguments
+        [BestFriend]
+        internal sealed class TaggedOptions
         {
             [Argument(ArgumentType.Multiple, HelpText = "New column definition(s) (optional form: name:srcs)",
                 Name = "Column", ShortName = "col", SortOrder = 1)]
             public TaggedColumn[] Columns;
         }
 
-        public sealed class ColumnInfo
+        [BestFriend]
+        internal sealed class ColumnInfo
```

Comment:
>ColumnInfo [](start = 30, length = 10)

I think we want to leave this public, because it is one of the arguments of the extensions.

Reply:
That class actually contains the notion of 'alias' which is purely for internal use (in another transform). I think it should stay internal. I do agree that we should make the Columns property public though and I will do that in my next iteration. 

---
In reply to: [253123275](https://github.com/dotnet/machinelearning/pull/2363#discussion_r253123275) [](ancestors = 253123275)